alica_engine/src/engine/AlicaTimer.cpp

```diff
@@ -0,0 +1,78 @@
+#include "engine/AlicaTimer.h"
+
+#include <thread>
+
+namespace alica
+{
+
+class AlicaSystemTimer::AlicaSystemTimerImpl
+{
+public:
+    AlicaSystemTimerImpl(TimerCb&& userCb, alica::AlicaTime period)
+            : _userCb(std::move(userCb))
+            , _period(period.inMilliseconds())
+            , _isActive(false)
+    {
+    }
+
+    ~AlicaSystemTimerImpl()
+    {
+        if (_isActive) {
+            stop();
+        }
+    }
+
+    void start()
+    {
+        if (_isActive) {
+            stop();
+        }
+        _isActive = true;
+        _thread = std::thread([this]() {
+            using namespace std::chrono;
+            auto inner_loop_duration = milliseconds(std::min(int64_t(500), _period));
+            while (_isActive) {
+                auto now = system_clock::now().time_since_epoch();
+                _userCb();
```

Comment:
Don't we want to do this after sleeping?

Reply:
Not if we want to retain current behaviour which is that the first run call occurs immediately after init()